modules/library/src/main/scala/zio/elasticsearch/executor/response/AggregationResponse.scala

```diff
@@ -350,107 +358,13 @@ private[elasticsearch] final case class TermsAggregationBucket(
   subAggregations: Option[Map[String, AggregationResponse]] = None
 ) extends AggregationBucket
 
-private[elasticsearch] object TermsAggregationBucket extends JsonDecoderOps {
+private[elasticsearch] object TermsAggregationBucket extends BucketDecoder {
   implicit val decoder: JsonDecoder[TermsAggregationBucket] = Obj.decoder.mapOrFail { case Obj(fields) =>
-    val allFields = fields.flatMap { case (field, data) =>
-      field match {
-        case "key" =>
-          Some(field -> data.toString.replaceAll("\"", ""))
-        case "doc_count" =>
-          Some(field -> data.unsafeAs[Int])
-        case _ =>
-          val objFields = data.unsafeAs[Obj].fields.toMap
+    val allFields = BucketDecoder(fields).allFields
```

Comment:
Because we are using `BucketDecoder` this way, maybe we don't even need this class to be `implicit` and then there is no need for `sealed trait`.

Reply:
If we do this, we should probably revert changes about `JsonDecoderOps`.